src/NuGet.Core/NuGet.ProjectModel/Utf8JsonReaderExtensions.cs

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Text.Json;
+
+namespace NuGet.ProjectModel
+{
+    internal static class Utf8JsonReaderExtensions
+    {
+        internal static string ReadTokenAsString(this ref Utf8JsonReader reader)
```

Comment:
I am curious about the use of the `ref` keyword for the reader parameter. My understanding is that reference types are already passed by reference, so I'm not sure of the benefit of using ref. Please correct me if my understanding is off.